pkg/virt-handler/hotplug-disk/mount.go

```diff
@@ -635,26 +635,29 @@ func (m *volumeMounter) Unmount(vmi *v1.VirtualMachineInstance, cgroupManager cg
 			}
 			return err
 		}
-		for _, volumeStatus := range vmi.Status.VolumeStatus {
-			if volumeStatus.HotplugVolume == nil {
+		// ideally we soud be looking at the disks in the domain
```

Comment:
```suggestion
		// ideally we should be looking at the disks in the domain
```

Reply:
No easy way to get the domain here?

Reply:
It's totally possible but the mounter is blissfully unaware of the domain at the moment and I think it would be nice to keep it that way.

But if we assume the following we should be good:

1. A disk removed from vmi spec will get removed from the domain
2. Domain is synced before unmount
3. Unmount won't get called unless domain sync succeeds

What do you think @awels?

Reply:
Sounds good to me.